operator/apis/lifecycle/v1alpha2/keptnworkload_types.go

```diff
@@ -102,10 +102,18 @@ func (w KeptnWorkload) GenerateWorkloadInstance(previousVersion string, traceCon
 	}
 }
 
-func (i KeptnWorkload) GetSpanAttributes() []attribute.KeyValue {
+func (w KeptnWorkload) GetSpanAttributes() []attribute.KeyValue {
 	return []attribute.KeyValue{
-		common.AppName.String(i.Spec.AppName),
-		common.WorkloadName.String(i.Name),
-		common.WorkloadVersion.String(i.Spec.Version),
+		common.AppName.String(w.Spec.AppName),
```

Comment:
afaik, `AppVersion` is always defined. Hence, I suggest also adding this info. Similarly, I would do the same for Event annotation.

Reply:
We don't have this information in the Workloads/WorkloadInstances, should we add it there ? It will be a breaking change and might be problematic to add it directly to resource, as we do not have this information during creation of Workloads. We can of course retrieve it in runtime, but that's additional logic, API calls etc. We only know the appName when creating the Workloads (in mutating pod we get it from annotations)